### PR TITLE
Quiz

### PR DESCRIPTION
**Pull request by mattbruton**
First pass at SPA quiz. Was a little shaky on the requested technical specs function that takes an element and a color as an argument, but if you look around at that specific function, it does take two arguments.

### CODE REVIEW

File: quiz.js [hunk 1]
```diff
@@ -1,57 +1,57 @@
...
+            carString = `<div class="col-sm-4 car" style="border: 3px solid ${inventory[i].color};">`;
+            carString += `<h1>${inventory[i].make}</h1>`;
+            carString += `<h2>${inventory[i].model}</h2>`;
+            carString += `<h4>Year: ${inventory[i].year}</h4>`;
+            carString += `<h4>Price: $${inventory[i].price}</h4>`;
+            carString += `<p>${inventory[i].description}</p>`;
+            carString += `<img class="blankImg" src="blank.png">`;
+            carString += `</div>`;
+            htmlContainer.innerHTML += carString;
+        };
```
Comment (JoeShep) [inline quiz.js]: This section is not very DRY. If you set a variable to hold either an empty string, "",  or the string "sold", based on whether it's true or not, you can then insert that variable into the "class" on line 16. Then the entire "else" section is unnecessary.

File: colorBorderIIFE.js [hunk 2]
```diff
@@ -1,73 +1,70 @@
...
-        }
-    })
-  };
+    newCarLot.styleChanger = function(element, color) {
+        container.addEventListener("click", function(event) {
+            var carDiv = document.getElementsByClassName('car');
+            for (var i = 0; i < carDiv.length; i++) {
+                if (carDiv[i].classList.contains(element)) {
+                    carDiv[i].style.borderWidth = "thick";
+                    carDiv[i].style.backgroundColor = color;
```
Comment (JoeShep) [inline colorBorderIIFE.js]: I would prefer to see all the style changes happen by adding a class that has these styles attached to it in the CSS.

**Comment by mattbruton** [comment]
Sure, should be able to knock those changes out tonight. Thanks for looking it over.

**Comment by JoeShep** [comment]
Matt, you have this working well, which is great. Nice job. But I would prefer that you refactor `eventiife.js`  and `colorBorderIIFE.js` to separate out the logic and the event listener attachment. Ideally, the colorBorder file would contain only the logic for changing/resetting the border and background, while the event listener file would attach event listeners that called the functions defined in the border color file. Come chat if this is confusing. Happy to go over it with you.

**Comment by mattbruton** [comment]
No problem, doesn't seem like it will be too much of an issue.

**Comment by mattbruton** [comment]
So far fixed (I think...) two of the issues you had with my quiz. Working through the part where you wanted me to change the styling of selected/non-selected cars using a class instead of applying inline CSS. I know how to do what you're asking, but it isn't working the way I want it to currently. Gonna tinker around with it throughout the day and should have something working later.

**Comment by mattbruton** [comment]
Alright, so the styling should only be affected by the CSS file, outside of selecting the border color of each card, since I don't know how to do that any other way than what I have with it being pulled from the JSON file. 

I made the changes to the functions you asked, but now the quiz doesn't pass the last step of the specs. I don't know how to pass the element/color without using inline styling as I can't change the values in my CSS file through JS.

**Comment by mattbruton** [comment]
Joe said :+1: in person.

**Pull request merged**